backend/api/Services/MissionSchedulingService.cs

```diff
@@ -543,5 +546,40 @@ private static float CalculateDistance(Pose pose1, Pose pose2)
         public static event EventHandler<RobotAvailableEventArgs>? RobotAvailable;
         protected virtual void OnMissionCompleted(MissionCompletedEventArgs e) { MissionCompleted?.Invoke(this, e); }
         public static event EventHandler<MissionCompletedEventArgs>? MissionCompleted;
+
+        public async Task<Robot> MissionPreCheck(string robotId)
+        {
+            var robot = await robotService.ReadById(robotId);
+
+            if (robot is null)
+            {
+                string errorMessage = $"The robot with ID {robotId} could not be found";
+                logger.LogError("{Message}", errorMessage);
+                throw new RobotNotFoundException(errorMessage);
+            }
+
+            if (!robot.IsRobotPressureTooLow())
+            {
+                string errorMessage = $"The robot pressure on {robot.Name} is too low to start a mission";
+                logger.LogError("{Message}", errorMessage);
+                throw new RobotPreCheckFailedException(errorMessage);
+            }
+
+            if (!robot.IsRobotPressureTooHigh())
+            {
+                string errorMessage = $"The robot pressure on {robot.Name} is too high to start a mission";
+                logger.LogError("{Message}", errorMessage);
+                throw new RobotPreCheckFailedException(errorMessage);
+            }
+
+            if (!robot.IsRobotBatteryTooLow())
+            {
+                string errorMessage = $"The robot battery level on {robot.Name} is too low to start a mission";
+                logger.LogError("{Message}", errorMessage);
+                throw new RobotPreCheckFailedException(errorMessage);
+            }
+
+            return robot;
```

Comment:
Why does the function return the robot?

Reply:
We might want to rename it to something where it would make sense for it to return the robot (eg. GetRobotWithPreCheck), otherwise we should not return the robot and just get the robot again in the controller function.

Reply:
It returns a robot because the endpoints in MissionSchedulingController.cs use the robot object after the pre-checks